library/core/src/iter/adapters/array_windows.rs

```diff
@@ -0,0 +1,102 @@
+use crate::iter::{ExactSizeIterator, Fuse, FusedIterator, Iterator, TrustedLen};
+
+use crate::array;
+
+/// An iterator over all contiguous windows of length `N`. The windows overlap.
+/// If the iterator is shorter than `N`, the iterator returns no values.
+///
+/// This `struct` is created by [`Iterator::array_windows`]. See its
+/// documentation for more.
+#[derive(Debug, Clone)]
+#[must_use = "iterators are lazy and do nothing unless consumed"]
+#[unstable(feature = "iter_array_windows", reason = "recently added", issue = "none")]
+pub struct ArrayWindows<I, const N: usize>
+where
+    I: Iterator,
+    I::Item: Clone,
+{
+    iter: Fuse<I>,
+    last: Option<[I::Item; N]>,
+}
+
+impl<I, const N: usize> ArrayWindows<I, N>
+where
+    I: Iterator,
+    I::Item: Clone,
+{
+    #[inline]
+    pub(in crate::iter) fn new(iter: I) -> Self {
+        assert!(N != 0);
```

Comment:
Is there some way to assert it in compile time?

Reply:
https://blog.rust-lang.org/2021/12/02/Rust-1.57.0.html#panic-in-const-contexts

Reply:
It would be ideal but I couldn't get it to work. I tried it for #92393 and get the following error

```text
error[E0401]: can't use generic parameters from outer function
  --> library/core/src/iter/adapters/array_chunks.rs:86:31
   |
81 | impl<I, const N: usize> ArrayChunks<I, N>
   |               - const parameter from outer function
...
86 |         const _: () = assert!(N != 0, "chunk size must be non-zero");
   |                               ^ use of generic parameter from outer function

For more information about this error, try `rustc --explain E0401`.
```

And if I make the const part of the impl block it doesn't result in a compile time error.